Error: The terraform-provider-datadog_v3.32.0 plugin crashed!

### Datadog Terraform Provider Version

v3.32.0

### Terraform Version

v1.6.3

### What resources or data sources are affected?

datadog_dashboard

### Terraform Configuration Files

_No response_

### Relevant debug or panic output

datadog_dashboard.example: Modifying... [id=8fn-epp-u6n]
╷
│ Error: Request cancelled
│
│   with datadog_dashboard.example,
│   on main.tf line 6, in resource "datadog_dashboard" "example":
│    6: resource "datadog_dashboard" "example" {
│
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

Stack trace from the terraform-provider-datadog_v3.32.0 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 129 [running]:
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogApmOrLogQuery(0x103a44460?)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:9105 +0x818
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogTimeseriesRequests(0x14001062230)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:7347 +0x260
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogTimeseriesDefinition(0x103a44460?)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:5996 +0x9c
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogWidget(0x103bc9300?)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:1214 +0xe10
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogWidgets(0x14001062628)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:1103 +0xcc
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogDashboard(0x14000f22680)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:374 +0x46c
github.com/terraform-providers/terraform-provider-datadog/datadog.resourceDatadogDashboardUpdate({0x103d0d3e0?, 0x14001ca8ff0?}, 0x14000f22680, {0x103934180?, 0x140005f0ff0})
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:202 +0x10c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0x1400073aa80, {0x103d0d3e0, 0x14001ca8ff0}, 0xd?, {0x103934180, 0x140005f0ff0})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/resource.go:767 +0xe8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x1400073aa80, {0x103d0d3e0, 0x14001ca8ff0}, 0x14001142dd0, 0x14000f39d00, {0x103934180, 0x140005f0ff0})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/resource.go:874 +0x6b0
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x1400000d3e0, {0x103d0d3e0?, 0x14001ca8ed0?}, 0x1400070e7d0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/grpc_provider.go:1024 +0xb78
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ApplyResourceChange({0x1400070c960, 0x1400070c9c0, {0x14000e6bb40, 0x2, 0x2}, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...}, ...)
	github.com/hashicorp/terraform-plugin-mux@v0.8.0/tf5muxserver/mux_server_ApplyResourceChange.go:27 +0xdc
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x14000e96640, {0x103d0d3e0?, 0x14001ca8180?}, 0x1400209a070)
	github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov5/tf5server/server.go:821 +0x3bc
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x103c35220?, 0x14000e96640}, {0x103d0d3e0, 0x14001ca8180}, 0x1400209a000, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:422 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000ed01e0, {0x103d12a00, 0x14000ffcd00}, 0x14002088000, 0x14000f53e60, 0x104a05f00, 0x0)
	google.golang.org/grpc@v1.56.3/server.go:1335 +0xc90
google.golang.org/grpc.(*Server).handleStream(0x14000ed01e0, {0x103d12a00, 0x14000ffcd00}, 0x14002088000, 0x0)
	google.golang.org/grpc@v1.56.3/server.go:1712 +0x82c
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.56.3/server.go:947 +0xb4
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.56.3/server.go:958 +0x174

Error: The terraform-provider-datadog_v3.32.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

### Expected Behavior

Should able to update or create the resource 

### Actual Behavior

terraform apply and terraform apply -auto-approve are failing with 
`Error: The terraform-provider-datadog_v3.32.0 plugin crashed!`

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

--- a/datadog/resource_datadog_dashboard.go
+++ b/datadog/resource_datadog_dashboard.go
@@ -9102,20 +9102,21 @@ func buildDatadogApmOrLogQuery(terraformQuery map[string]interface{}) *datadogV1
 	if terraformGroupBys, ok := terraformQuery["group_by"].([]interface{}); ok && len(terraformGroupBys) > 0 {
 		datadogGroupBys := make([]datadogV1.LogQueryDefinitionGroupBy, len(terraformGroupBys))
 		for i, g := range terraformGroupBys {
-			groupBy := g.(map[string]interface{})
-			// Facet
-			datadogGroupBy := datadogV1.NewLogQueryDefinitionGroupBy(groupBy["facet"].(string))
-			// Limit
-			if v, ok := groupBy["limit"].(int); ok && v != 0 {
-				datadogGroupBy.SetLimit(int64(v))
-			}
-			// Sort
-			if sortList, ok := groupBy["sort_query"].([]interface{}); ok && len(sortList) > 0 {
-				if sort, ok := sortList[0].(map[string]interface{}); ok && len(sort) > 0 {
-					datadogGroupBy.Sort = buildDatadogGroupBySort(sort)
+			if groupBy, ok := g.(map[string]interface{}); ok {
+				// Facet
+				datadogGroupBy := datadogV1.NewLogQueryDefinitionGroupBy(groupBy["facet"].(string))
+				// Limit
+				if v, ok := groupBy["limit"].(int); ok && v != 0 {
+					datadogGroupBy.SetLimit(int64(v))
 				}
+				// Sort
+				if sortList, ok := groupBy["sort_query"].([]interface{}); ok && len(sortList) > 0 {
+					if sort, ok := sortList[0].(map[string]interface{}); ok && len(sort) > 0 {
+						datadogGroupBy.Sort = buildDatadogGroupBySort(sort)
+					}
+				}
+				datadogGroupBys[i] = *datadogGroupBy
 			}
-			datadogGroupBys[i] = *datadogGroupBy
 		}
 		datadogQuery.SetGroupBy(datadogGroupBys)
 	}

